libs/androidlib/src/mill/androidlib/AndroidNativeAppModule.scala

```diff
@@ -0,0 +1,103 @@
+package mill.androidlib
+
+import mill.T
+import mill.define.{PathRef, Task}
+
+trait AndroidNativeAppModule extends AndroidAppModule {
+
+  /**
+   * The path of each of the C/C++ source files to be compiled.
+   */
+  def androidExternalNativeLibs: T[Seq[PathRef]]
+
+  /**
+   * All the different ABI's that are supported by the Android SDK.
+   */
+  def androidSupportedAbis: T[Seq[String]] = Task {
+    Seq("armeabi-v7a", "arm64-v8a", "x86", "x86_64")
+  }
+
+  /**
+   * The version of the Android build tools to be used
+   */
+  def androidBuildToolsVersion: T[String] = Task {
+    androidSdkModule().buildToolsVersion().split('.')(0)
+  }
+
+  /**
+   * The name of the native library to be compiled.
+   */
+  def androidNativeLibName: T[String] = Task {
+    "libnative-lib"
+  }
+
+  /**
+   * This method compiles all the native libraries using CMake and Ninja
+   * and generates the .so files in the output directory for each of the supported ABI's.
+   */
```

Comment:
Are there any docs we can link to for this, or external examples? That would help anyone reading this figure out why this particular combination of flags is necessary

Reply:
added the link!